lib/rules/file-extension-in-import.js

```diff
@@ -8,7 +8,7 @@ const path = require("path")
 const fs = require("fs")
 const mapTypescriptExtension = require("../util/map-typescript-extension")
 const visitImport = require("../util/visit-import")
-const packageNamePattern = /^(?:@[^/\\]+[/\\])?[^/\\]+$/u
+const packageNamePattern = /^(?:@[^/\\]+[/\\])?(?:[^./\\]+[/\\])*[^/\\]+$/u
```

Comment:
good to have a comment to explain the regex - regex is always easy to write but hard to read :)

Reply:
Will do!
FYI old regex was matching:
`optional @scope/ followed by non-slash characters`
which excluded imports like @scope/pkg/dir
new one:
`optional @scope/ followed by an optional number of groups containing 1 or more characters ending with a slash followed by by non-slash characters`

Reply:
Here is a quick before and after using my favourite regex renderer 😄

Before: [`/^(?:@[^/\\]+[/\\])?[^/\\]+$/u`](https://jex.im/regulex/#!flags=&re=%5E(%3F%3A%40%5B%5E%2F%5C%5C%5D%2B%5B%2F%5C%5C%5D)%3F%5B%5E%2F%5C%5C%5D%2B%24)
<img width="667" alt="image" src="https://github.com/eslint-community/eslint-plugin-n/assets/2230835/74f57d12-bfee-49a7-b838-00cf240e63d7">

After: [`/^(?:@[^/\\]+[/\\])?(?:[^./\\]+[/\\])*[^/\\]+$/u`](https://jex.im/regulex/#!flags=&re=%5E(%3F%3A%40%5B%5E%2F%5C%5C%5D%2B%5B%2F%5C%5C%5D)%3F(%3F%3A%5B%5E.%2F%5C%5C%5D%2B%5B%2F%5C%5C%5D)*%5B%5E%2F%5C%5C%5D%2B%24)
<img width="917" alt="image" src="https://github.com/eslint-community/eslint-plugin-n/assets/2230835/7889b79a-4d72-4939-beae-77ea0116e06f">

It does looks like the change should work too:
<img width="1601" alt="image" src="https://github.com/eslint-community/eslint-plugin-n/assets/2230835/292fb298-4f2f-47c3-8035-18fa9538ae90">

<img width="1601" alt="image" src="https://github.com/eslint-community/eslint-plugin-n/assets/2230835/ba2fb7fa-08b4-44a7-a7d1-777ccb80b9f6">